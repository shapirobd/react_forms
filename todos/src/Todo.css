.Todo-remove-btn {
	text-align: right;
}
.Todo-edit-btn {
	text-align: right;
	margin-right: 10px;
}
.Todo-complete-btn {
	text-align: right;
}

.Todo-edit-form {
	width: 100%;
	background-color: #e4e0de;
}
