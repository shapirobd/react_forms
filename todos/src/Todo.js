import React, { useState } from "react";
import EditTodoForm from "./EditTodoForm";

const Todo = ({ task, removeTodo, editTodo, completeTodo, completed }) => {
	const EDIT_FORM_INITIAL_STATE = "none";
	const COMPLETE_BTN_INITIAL_STATE = "Mark as Completed";
	const TODO_STRIKE_INITIAL_STATE = "none";
	const [editFormVisibility, setEditFormVisibility] = useState(
		EDIT_FORM_INITIAL_STATE
	);
	const [completeBtnText, setCompleteBtnText] = useState(
		COMPLETE_BTN_INITIAL_STATE
	);
	const [todoStrike, setTodoStrike] = useState(TODO_STRIKE_INITIAL_STATE);
	const toggleEditFormVisibility = () => {
		setEditFormVisibility((editFormVisibility) =>
			editFormVisibility === "none" ? "block" : "none"
		);
	};
	const toggleCompleteBtnText = () => {
		setCompleteBtnText((completeBtnText) =>
			completeBtnText === "Mark as Completed"
				? "Mark as Incomplete"
				: "Mark as Completed"
		);
	};
	const toggleTodoStrike = () => {
		setTodoStrike((todoStrike) =>
			todoStrike === "none" ? "line-through" : "none"
		);
	};
	const handleCompleteBtnClick = () => {
		toggleTodoStrike();
		toggleCompleteBtnText();
	};
	return (
		<div className="Todo">
			<p style={{ textDecoration: todoStrike }}>{task}</p>
			<button onClick={removeTodo}>X</button>
			<button onClick={toggleEditFormVisibility}>Edit</button>
			<button onClick={handleCompleteBtnClick}>{completeBtnText}</button>
			<EditTodoForm
				toggleEditFormVisibility={toggleEditFormVisibility}
				visibility={editFormVisibility}
				editTodo={editTodo}
				originalTodo={task}
			/>
		</div>
	);
};

export default Todo;
