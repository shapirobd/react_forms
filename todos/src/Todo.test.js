import { render, screen } from "@testing-library/react";
import Todo from "./Todo";

it("renders without crashing", () => {
	render(<Todo />);
});

it("matches snapshot", () => {
	const { asFragment } = render(<Todo />);
	expect(asFragment()).toMatchSnapshot();
});

it("contains the given task text", () => {
	const { queryByText } = render(<Todo task="Do dishes" />);
	expect(queryByText("Do dishes")).toBeInTheDocument();
});
