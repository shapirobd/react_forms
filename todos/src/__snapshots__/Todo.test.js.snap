// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`has a p element that contains task 1`] = `
<DocumentFragment>
  <div
    class="Todo"
  >
    <p
      class="Todo-task"
      style="text-decoration: none;"
    />
    <button
      class="Todo-edit-btn"
    >
      Edit
    </button>
    <button
      class="Todo-remove-btn"
    >
      X
    </button>
    <button
      class="Todo-complete-btn"
    >
      Mark as Completed
    </button>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Todo"
  >
    <p
      class="Todo-task"
      style="text-decoration: none;"
    />
    <button
      class="Todo-edit-btn"
    >
      Edit
    </button>
    <button
      class="Todo-remove-btn"
    >
      X
    </button>
    <button
      class="Todo-complete-btn"
    >
      Mark as Completed
    </button>
  </div>
</DocumentFragment>
`;
